.github/workflows/nightly-test-release.yaml

```diff
@@ -0,0 +1,126 @@
+name: Test release process nightly
+
+on:
+  schedule:
+    - cron: "0 0 * * *" # Run every day at midnight (UTC)
+  workflow_dispatch: # Allow running manually on demand
+
+env:
+  RELEASE_TAG: t9.9.9-fake
+
+jobs:
+  nightly-test-release:
+    name: Tag current version for testing
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: main
+          fetch-depth: 0
+      - name: Set and push fake tag for release
+        run: |
+          git tag ${{ env.RELEASE_TAG }}
+      - name: Push changes
+        uses: ad-m/github-push-action@master
+        with:
+          tags: true
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+  build-push-services:
+    name: Build, sign and provenance
+    needs: [nightly-test-release]
+    permissions:
+      actions: read
+      contents: write
+      packages: write
+      id-token: write
+    strategy:
+      matrix:
+        destination: [ghcr]
```

Comment:
I think it is possible to replicate prod destination here by using a different set of secrets, and pointing to ghcr repo. Password could be a new token with sufficient permissions. This way we will use single registry but ensure that both workflow scenarios pass without issues.